aiohomekit/controller/ip/connection.py

```diff
@@ -545,9 +551,6 @@ async def _reconnect(self):
                     # Authentication errors should bubble up because auto-reconnect is unlikely to help
                     raise
 
-                except asyncio.CancelledError:
-                    return
-
```

Comment:
Newer python changed CancelledError to inherit from base exception so we should not catch it here and let it raise instead otherwise this leads to odd behavior since it blocks cancellation but doesn't decrement the cancellation counter with uncancel

Reply:
I need to check this since I think part of the reason it was added was to prevent cancelation from bubbling up on anything that awaits the task.  We likely need to shield higher up for that though